### ISSUE TITLE
build: Use Xcode .xip as Gitian input

### ISSUE THREAD

**Issue by dongcarl** [bitcoin/bitcoin#18674]
It seems to me that the [manual extraction](https://github.com/bitcoin/bitcoin/tree/master/contrib/macdeploy#sdk-extraction) of `MacOSX10.*.sdk.tar.gz` can be a source of fragility, confusion, and non-determinism.

Perhaps what we could do going forward is to have the unmodified Xcode `.xip` as the Gitian input, and perform the extraction inside the Gitian environment. This has the additional benefit of the `.xip` being trivially deterministic (and thus easy to compare) as long as people are downloading the same version.

**Comment by fanquake** [comment]
Concept ACK. The macOS SDK seems to be the step that causes the most trouble for gitian builders. Whether that's actually getting it, putting it in the right place, ensuring it's named correctly (zipped and after unzipping) etc.

**Comment by MarcoFalke** [comment]
Will this be added as a script to `./depends`, so that it is properly cached and only generated once per major release?

**Comment by jasdeep9899** [comment]
I am doing a project and for that I need minimum 150 btc testnet coins... Can anyone help me ... My btc testnet wallet address is

mma9KKAAu32M5fJMchMoZJNB1xBJxJ5Z8c

**Comment by Sjors** [comment]
Concept ACK. That also removes any potential difference between macOS and linux users in the way they produce the SDK archive.

**Comment by jonasschnelli** [comment]
I guess the only downside is that one needs to provide a much larger file (a few GBs)?

**Comment by dongcarl** [comment]
> I guess the only downside is that one needs to provide a much larger file (a few GBs)?

Right, but you needed to download in the first place anyway, and the increase in size is not too bad compared to the rest of Gitian

**Comment by jonasschnelli** [comment]
> > I guess the only downside is that one needs to provide a much larger file (a few GBs)?

> Right, but you needed to download in the first place anyway, and the increase in size is not too bad compared to the rest of Gitian

I thought more in the context of CI systems where either the "cached" .xip needs to be loaded over (like to a VM) or the .xip needs to be downloaded form some sort of a remote system (like travis loads the MacOS SDK from bitcoincore.org.).

**Comment by hebasto** [comment]
Is this still relevant after migration to Guix?

**Comment by fanquake** [comment]
> Is this still relevant after migration to Guix?

Yes. The issue this is solving is the same for Guix and gitian.

**Comment by prusnak** [comment]
Related https://github.com/bitcoin/bitcoin/pull/24534 - make macOS headers extraction from XIP file deterministic.
